qiita_db/util.py

```diff
@@ -2311,3 +2317,318 @@ def send_email(to, subject, body):
         raise RuntimeError("Can't send email!")
     finally:
         smtp.close()
+
+
+def resource_allocation_plot(file, cname, sname, col_name):
+    """Builds resource allocation plot for given filename and jobs
+
+    Parameters
+    ----------
+    file : str, required
+        Builds plot for the specified file name. Usually provided as tsv.gz
+    cname: str, required
+        Specified job type
+    sname: str, required
+        Specified job sub type.
+    col_name: str, required
+        Specifies x axis for the graph
+
+    Returns
+    ----------
+    matplotlib.pyplot object
+        Returns a matplotlib object with a plot
+    """
+
+    df = pd.read_csv(file, sep='\t', dtype={'extra_info': str})
+    df['ElapsedRawTime'] = pd.to_timedelta(df.ElapsedRawTime)
+    df = df[(df.cName == cname) & (df.sName == sname)]
+    df.dropna(subset=['samples', 'columns'], inplace=True)
+    df[col_name] = df.samples * df['columns']
+    df[col_name] = df[col_name].astype(int)
+
+    fig, axs = plt.subplots(ncols=2, figsize=(10, 4), sharey=False)
+
+    ax = axs[0]
+    # models for memory
+    mem_model1 = (lambda x, k, a, b: k * np.log(x) + x * a + b)
```

Comment:
It's not desirable to have these defined once in utils and once in tests. If you need them for tests, I would suggest moving them out of this function and defining them at the top of util.py. Then you can still use them in this function, but you can also import them into test_util.py (actually I think you just need to import the model_mem and model_time arrays) and use them there. In that way you're treating util.py as the place to go to define and get that information. Currently you have to know that this is defined in two places and make changes to both as needed.